ruby/ruby/IceLocal/LocalException.rb

```diff
@@ -3,635 +3,454 @@
-            end
+    class FacetNotExistException < ::Ice::RequestFailedException
+        def initialize(id = ::Ice::Identity.new, facet = '', operation = '')
+            super(id, facet, operation)
         end
 
-        T_FacetNotExistException = ::Ice::__defineException('::Ice::FacetNotExistException', FacetNotExistException, ::Ice::T_RequestFailedException, [])
+        def to_s
+            '::Ice::FacetNotExistException'
+        end
     end
 
-    if not defined?(::Ice::OperationNotExistException)
-        class OperationNotExistException < ::Ice::RequestFailedException
-            def initialize(id=::Ice::Identity.new, facet='', operation='')
-                super(id, facet, operation)
-            end
-
-            def to_s
-                '::Ice::OperationNotExistException'
-            end
+    class OperationNotExistException < ::Ice::RequestFailedException
+        def initialize(id = ::Ice::Identity.new, facet = '', operation = '')
+            super(id, facet, operation)
         end
 
-        T_OperationNotExistException = ::Ice::__defineException('::Ice::OperationNotExistException', OperationNotExistException, ::Ice::T_RequestFailedException, [])
+        def to_s
+            '::Ice::OperationNotExistException'
+        end
     end
 
-    if not defined?(::Ice::SyscallException)
-        class SyscallException < Ice::LocalException
-            def initialize(error=0)
-                @error = error
-            end
-
-            def to_s
-                '::Ice::SyscallException'
-            end
+    class SyscallException < Ice::LocalException
+        def initialize(error = 0)
+            @error = error
+        end
 
-            attr_accessor :error
+        def to_s
+            '::Ice::SyscallException'
         end
 
-        T_SyscallException = ::Ice::__defineException('::Ice::SyscallException', SyscallException, nil, [["error", ::Ice::T_int, false, 0]])
+        attr_accessor :error
     end
 
-    if not defined?(::Ice::SocketException)
-        class SocketException < ::Ice::SyscallException
-            def initialize(error=0)
-                super(error)
-            end
-
-            def to_s
-                '::Ice::SocketException'
-            end
+    class SocketException < ::Ice::SyscallException
+        def initialize(error = 0)
+            super(error)
         end
 
-        T_SocketException = ::Ice::__defineException('::Ice::SocketException', SocketException, ::Ice::T_SyscallException, [])
+        def to_s
+            '::Ice::SocketException'
+        end
     end
 
-    if not defined?(::Ice::CFNetworkException)
-        class CFNetworkException < ::Ice::SocketException
-            def initialize(error=0, domain='')
-                super(error)
-                @domain = domain
-            end
-
-            def to_s
-                '::Ice::CFNetworkException'
-            end
+    class CFNetworkException < ::Ice::SocketException
```

Comment:
isn't this only for iOS, can we remove it?

Reply:
Yes, I'll remove it in the refactoring PR.